lightning/src/events/mod.rs

```diff
@@ -1552,10 +1551,22 @@ impl Writeable for Event {
 			},
 			&Event::PaymentFailed { ref payment_id, ref payment_hash, ref reason } => {
 				15u8.write(writer)?;
+				let (payment_hash, invoice_received) = match payment_hash {
+					Some(payment_hash) => (payment_hash, Some(true)),
+					None => (&PaymentHash([0; 32]), Some(false)),
+				};
+				let legacy_reason = match reason {
+					Some(PaymentFailureReason::UnknownRequiredFeatures)
+						| Some(PaymentFailureReason::InvoiceRequestExpired)
+						| Some(PaymentFailureReason::InvoiceRequestRejected) => &None,
+					reason => reason,
+				};
 				write_tlv_fields!(writer, {
 					(0, payment_id, required),
-					(1, reason, option),
+					(1, legacy_reason, option),
 					(2, payment_hash, required),
+					(3, invoice_received, option),
```

Comment:
Doesn't matter much, but it would make more sense to write this as `required` since we always write it

Reply:
We should def make it `Option`al on read, but on writing it's fine to make it required since we always write it (there's no difference between `required` and an `option` where we always write).

Reply:
Ah, I see. Done.